app/tests/src/test_ingest_utils.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from unittest.mock import ANY, Mock
+
+from sqlalchemy import delete, select
+
+from src.db.models.document import Document
+from src.util.ingest_utils import _drop_existing_dataset, process_and_ingest_sys_args
+from tests.src.db.models.factories import DocumentFactory
+
+
+def test__drop_existing_dataset(db_session, enable_factory_create):
+    db_session.execute(delete(Document))
+
+    docs = DocumentFactory.create_batch(2)
```

Comment:
If there's a miniscule chance that the Documents are given the same dataset string, then to eliminate the risk of a flaky test, create the 2 documents with different `dataset` values.